### PR TITLE
[receiver/mongodbatlasreceiver]: add / rename a few attributes

### PR DESCRIPTION
**Pull request by mstumpfx**
**Description:** Adds a 3 new attributes that are useful identifiers for this receiver. Also renaming a few based on naming guidelines.

### CODE REVIEW

File: receiver/mongodbatlasreceiver/receiver.go [hunk 1]
```diff
@@ -90,6 +90,7 @@ func (s *receiver) poll(ctx context.Context, time timeconstraints) (pdata.Metric
 		}
 		for _, project := range projects {
 			resourceAttributes.InsertString("mongodb.atlas.project", project.Name)
+			resourceAttributes.InsertString("mongodb.atlas.project.id", project.ID)
```
Comment (dmitryax) [inline receiver/mongodbatlasreceiver/receiver.go:93]: This goes against the [naming guidelines](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md#attribute-naming)

> Names SHOULD NOT coincide with namespaces

Maybe `mongodb_atlas.project.name` and `mongodb_atlas.project.name`?

File: receiver/mongodbatlasreceiver/receiver.go [hunk 2]
```diff
@@ -99,6 +100,8 @@ func (s *receiver) poll(ctx context.Context, time timeconstraints) (pdata.Metric
 				resourceAttributes.CopyTo(resource.Attributes())
 				resource.Attributes().InsertString("host.name", process.Hostname)
 				resource.Attributes().InsertString("process.port", strconv.Itoa(process.Port))
+				resource.Attributes().InsertString("process.typename", process.TypeName)
```
Comment (dmitryax) [inline receiver/mongodbatlasreceiver/receiver.go:103]: Is this usually being used as one word? If not, maybe not`process.type_name` instead?

File: receiver/mongodbatlasreceiver/receiver.go (hunk 1 above)
Comment (mstumpfx) [reply receiver/mongodbatlasreceiver/receiver.go:93]: Thanks @dmitryax, let me read through this again and update.

**Review by dmitryax — APPROVED** [review]
LGTM. Please rebase to resolve README conflicts

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 - Upgrade to use semantic conventions 1.6.1 (#7926)
 - `dynatraceexporter`: Validate QueueSettings and perform config validation in Validate() instead (#8020)
 - `signalfxexporter`: Add validation for `sending_queue` setting (#8026)
+- `mongodbatlasreceiver`: modify attributes to adhere to naming guidelines and add 3 new attributes (#7960)
```
Comment (dmitryax) [inline CHANGELOG.md:11]: I think this should be listed as a breaking change

**Review by codeboten — COMMENTED** [review]
Is the idea behind the change that atlas isn't a component of mongodb, therefore the  namespace `mongodb_atlas` makes more sense to represent it than `mongodb.atlas`?

**Comment by dmitryax** [comment]
@codeboten [MongoDB Atlas](https://www.mongodb.com/basics/mongodb-atlas-tutorial) is a cloud managed MongoDB, so it's not a component of MongoDB.

**Comment by dmitryax** [comment]
@zenmoto PTAL as an author if you have a chance

File: CHANGELOG.md (hunk 3 above)
Comment (mstumpfx) [reply CHANGELOG.md]: You right

File: CHANGELOG.md [hunk 4]
```diff
@@ -18,6 +18,7 @@
 ### 🛑 Breaking changes 🛑
 
 - Remove deprecated functions from jaeger translator (#8032)
+- `mongodbatlasreceiver`: modify attributes to adhere to naming guidelines and add 3 new attributes (#7960)
```
Comment (codeboten) [inline CHANGELOG.md:21]:
```suggestion
- `mongodbatlasreceiver`: rename `mongodb.atlas.*` attributes to `mongodb_atlas.*` adhering to naming guidelines. Adding 3 new attributes (#7960)
```

**Review by codeboten — COMMENTED** [review]
One suggestion in the changelog to give users a path for moving forward.

**Comment by mstumpfx** [comment]
thanks @codeboten, much appreciated.

**Review by codeboten — APPROVED** [review]

**Pull request merged**